harness/tests/integration_cases/test_raft_flow_control.rs

```diff
@@ -175,3 +175,103 @@ fn test_msg_app_flow_control_recv_heartbeat() {
         r.read_messages();
     }
 }
+
+#[test]
+fn test_msg_app_flow_control_with_freeing_resources() {
+    let l = default_logger();
+    let mut r = new_test_raft(1, vec![1, 2, 3], 5, 1, new_storage(), &l);
+
+    r.become_candidate();
+    r.become_leader();
+
+    for (_, pr) in r.prs().iter() {
+        assert!(!pr.ins.buffer_is_allocated());
+    }
+
+    for i in 1..=3 {
+        // Force the progress to be in replicate state.
+        r.mut_prs().get_mut(i).unwrap().become_replicate();
+    }
+
+    r.step(new_message(1, 1, MessageType::MsgPropose, 1))
+        .unwrap();
+
+    for (&id, pr) in r.prs().iter() {
+        if id != 1 {
+            assert!(pr.ins.buffer_is_allocated());
+        }
+    }
+
+    /*
+    1: cap=0/start=0/count=0/buffer=[]
+    2: cap=256/start=0/count=1/buffer=[2]
```

Comment:
count should be 1?

Reply:
I am not sure why `MsgPropose` sends `2` to others, but it did.